hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/HederaLifecyclesImplTest.java

```diff
@@ -44,8 +44,11 @@
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
 
 @ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
```

Comment:
How much work would it be to avoid needing this? Ime lenient mocking can hide real problems.

Reply:
I was able to remove the mock public key and the stubbing that is being complained about in the `HederaLifecyclesTest.java` unit tests without problem and can remove the lenient marking in this file. 

When I do that for the `ServicesStateTest.java` I get an exception thrown on this [line of code](https://github.com/hashgraph/hedera-services/blob/c2ac301a51281361f8ac9517b19dc680e768d264/hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/ServicesStateTest.java#L944).   Somehow this code is requesting the `PublicKey` and is generating a null pointer exception.  If I mock the method retrieving the public key, the exception is not thrown, but mockito complains about an unused mock.  I am left scratching my head. 

One unit test accesses the code path of getting the public key while 3 do not.  This is a damned if you do and damned if you don't situation.  

This i my ugly solution, to add a boolean for when the method needs to be stubeed: 
```
 public static AddressBook createPretendBookFrom(final Platform platform, final boolean withKeyDetails, boolean mockPublicKey) {
        final var publicKey = mock(PublicKey.class);
        final var cert = mock(X509Certificate.class);
        if(mockPublicKey){
            Mockito.when(cert.getPublicKey()).thenReturn(publicKey);
        }
```

Reply:
Hmm interesting 😵‍💫 ...so not easy.